TestModels/Constraints/test/Helpers.dfy

```diff
@@ -3,160 +3,192 @@
-      var invalidStringLessThanOrEqualToTen: StringLessThanOrEqualToTen := invalidStringLessThanOrEqualToTenInput;
-      return invalidStringLessThanOrEqualToTen;
-    }
-
-    method InvalidMyBlob()
-      returns (invalid: MyBlob)
-    {
-      var invalidMyBlobInput := []; // Invalid because |x| < 1; predicate requires 1 <= |x| <= 10
-      assume {:axiom} IsValid_MyBlob(invalidMyBlobInput);
-      var invalidMyBlob: MyBlob := invalidMyBlobInput;
-      return invalidMyBlob;
-    }
-
-    // TODO: Write more invalid input generators...
+  import SimpleConstraints
+  import opened StandardLibrary.UInt
+  import opened SimpleConstraintsTypes
+  import opened Wrappers
+
+  // UTF-8 encoded "aws-kms"
+  const PROVIDER_ID: UTF8.ValidUTF8Bytes :=
+    var s := [0x61, 0x77, 0x73, 0x2D, 0x6B, 0x6D, 0x73];
+    assert UTF8.ValidUTF8Range(s, 0, 7);
+    s
+
+
+  // This returns a valid GetConstraintsInput object.
+  function method GetValidInput() : GetConstraintsInput
+  {
+    // var myComplexUniqueList :=
+    //   [ ComplexListElement(value := Some("one"), blob := Some([1, 1])),
+    //     ComplexListElement(value := Some("two"), blob := Some([2, 2]))
+    //   ];
+
+    GetConstraintsInput(
+      MyString := Some("bw1and10"),
+      NonEmptyString := Some("atleast1"),
+      StringLessThanOrEqualToTen := Some("leq10"),
+      MyBlob := Some([0, 1, 0, 1]),
+      NonEmptyBlob := Some( [0, 1, 0, 1]),
+      BlobLessThanOrEqualToTen := Some([0, 1, 0, 1]),
+      MyList := Some(["00", "11"]),
+      NonEmptyList := Some(["00", "11"]),
+      ListLessThanOrEqualToTen := Some(["00", "11"]),
+      MyMap := Some(map["0" := "1", "2" := "3"]),
+      NonEmptyMap := Some(map["0" := "1", "2" := "3"]),
+      MapLessThanOrEqualToTen := Some(map["0" := "1", "2" := "3"]),
+      // Alphabetic := Some("alphabetic"),
+      OneToTen := Some(3),
+      GreaterThanOne := Some(3),
+      LessThanTen := Some(3),
+      // MyUniqueList := Some(["one", "two"]),
+      // MyComplexUniqueList := Some(myComplexUniqueList),
+      MyUtf8Bytes := Some(PROVIDER_ID),
+      MyListOfUtf8Bytes := Some([PROVIDER_ID, PROVIDER_ID])
+    )
+  }
+
+  predicate method ValidInt32(x : int)
+  {
+    -0x8000_0000 <= x < 0x8000_0000
+  }
+
+  predicate method ValidInt64(x : int)
+  {
+    -0x8000_0000_0000_0000 <= x < 0x8000_0000_0000_0000
+  }
+
+  function method ForceUtf8Bytes(value : seq<uint8>) : Utf8Bytes
+  {
+    assume {:axiom} UTF8.ValidUTF8Seq(value);
+    assume {:axiom} IsValid_Utf8Bytes(value);
+    var myUtf8Bytes: Utf8Bytes := value;
+    myUtf8Bytes
+  }
+  
+  function method ForceListOfUtf8Bytes(value : seq<Utf8Bytes>) : ListOfUtf8Bytes
+  {
+    assume {:axiom} IsValid_ListOfUtf8Bytes(value);
+    var myListOfUtf8Bytes: ListOfUtf8Bytes := value;
+    myListOfUtf8Bytes
+  }
+
+  function method ForceLessThanTen(value : int) : LessThanTen
+  {
+    assume {:axiom} ValidInt32(value);
```

Comment:
We might want to loop in @robin-aws.
He pushed back on testing failure cases for SimpleConstraints using `assume` statements.